modules/azure_aks/main.tf

```diff
@@ -23,10 +23,10 @@ resource "azurerm_kubernetes_cluster" "aks" {
     # https://docs.microsoft.com/en-us/azure/aks/load-balancer-standard
     # https://docs.microsoft.com/en-us/azure/aks/egress-outboundtype
 
-    service_cidr       = var.aks_network_plugin == "kubenet" ? "10.0.0.0/16" : var.aks_service_cidr
-    dns_service_ip     = var.aks_network_plugin == "kubenet" ? "10.0.0.10" : var.aks_dns_service_ip
-    pod_cidr           = var.aks_network_plugin == "kubenet" ? "10.244.0.0/16" : null
-    docker_bridge_cidr = var.aks_network_plugin == "kubenet" ? "172.17.0.1/16" : var.aks_docker_bridge_cidr
+    service_cidr       = var.aks_service_cidr
+    dns_service_ip     = var.aks_dns_service_ip
+    pod_cidr           = var.aks_network_plugin == "kubenet" ? var.aks_pod_cidr : null
+    docker_bridge_cidr = var.aks_docker_bridge_cidr
```

Comment:
The end code may look like this but we need to add some extra logic here
```
service_cidr = var.aks_network_plugin == "kubenet" ? var.aks_service_cidr != "" ? var.aks_service_cidr : "10.0.0.0/16"
```
So we'll expand on this, but this is the right direction.

Reply:
Thanks for taking a look @thpang . I think the default would be covered from https://github.com/sassoftware/viya4-iac-azure/blob/main/modules/azure_aks/variables.tf#L83-L102 and https://github.com/sassoftware/viya4-iac-azure/blob/main/variables.tf#L126-L155 which is why I dind't include it in this line. Is there a situation where defaults wouldn't come from there?

Reply:
The defaults work, but having extra logic here guards against someone adding these variables and simply empty out the value. Then the default would not apply. Also these variables are not documented, so no real guidance.

Reply:
@thpang I am a bit new to terraform, but would doing a validation block in those files solve this as well?
https://github.com/sassoftware/viya4-iac-azure/blob/main/variables.tf#L140-L143

Or would you want the validation in the main.tf?

Reply:
The validation block would be at the variable level when its defined